core/commands/blame.py

```diff
@@ -238,9 +274,132 @@ def run(self, cursor_pt, edit):
+                        return commit
+                elif position == "newer":
+                    return log_commits[idx-1]
+
+    def newst_commit_for_file(self):
+        return self.git("log", "--format=%H", "--follow", "-n", "1", self.file_path).strip()
+
+    def find_line_and_open(self):
+        commit_hash = self.selected_commit_hash().strip()
+        if not commit_hash:
+            self.view.settings().set("git_savvy.commit_hash", self.newst_commit_for_file())
+        else:
+            self.view.settings().set("git_savvy.commit_hash", self.commit_before("older", commit_hash))
+
+        self.view.run_command("gs_blame_initialize_view")
+
+    def open(self, position):
+        settings = self.view.settings()
+        commit_hash = settings.get("git_savvy.commit_hash")
+        if not commit_hash:
+            settings.set("git_savvy.commit_hash", self.newst_commit_for_file())
+        else:
+            previous_commit_hash = self.commit_before(position, settings.get("git_savvy.commit_hash"))
+            settings.set("git_savvy.commit_hash", previous_commit_hash)
+        self.view.run_command("gs_blame_initialize_view")
+
+    def show_file_at_commit(self, from_line=False):
+        if from_line:
+            commit_hash = self.selected_commit_hash() or 'HEAD'
+        else:
+            commit_hash = self.view.settings().get("git_savvy.commit_hash", "HEAD")
+
+        self.view.window().run_command("gs_show_file_at_commit", {
+            "commit_hash": commit_hash,
+            "filepath": self.file_path,
+            "lineno": self.find_lineno(),
+            "lang" : self.view.settings().get('git_savvy.original_syntax', None)
+        })
+
+    def pick_new_commit(self):
+        self.view.run_command("gs_blame_pick_commit", {
+            "commit_hash": self.view.settings().get("git_savvy.commit_hash"),
+        })
+
+class GsBlamePickCommitCommand(TextCommand, GitCommand):
+
+    def run(self, *args, commit_hash=None):
+        self._file_path = self.file_path
+        self._branch = None
+        sublime.set_timeout_async(self.run_async)
+
+    def run_async(self):
+        settings = self.view.settings()
+        settings.set("git_savvy.commit_hash_old", settings.get("git_savvy.commit_hash"))
+        lp = BlameCommitPanel(
+            self.commit_generator(follow=True),
+            self.do_action,
+            )
+        lp.selected_commit(settings.get("git_savvy.commit_hash"))
+        lp.show()
+
+    def do_action(self, commit_hash):
+        settings = self.view.settings()
+        if commit_hash is None:
+            # Canceled panel
+            settings.set("git_savvy.commit_hash", settings.get("git_savvy.commit_hash_old"))
+            settings.erase("git_savvy.commit_hash_old")
+        else:
+            settings.set("git_savvy.commit_hash", commit_hash)
+        self.view.run_command("gs_blame_initialize_view")
+
+
+class BlameCommitPanel(PaginatedPanel):
```

Comment:
Check [this](https://github.com/divmain/GitSavvy/pull/661/commits/0f1c2e76e3cb8f457c471cf0fac39b9a077dce18)